src/wsproto/frame_protocol.py

```diff
@@ -569,7 +569,9 @@ def received_frames(self) -> Generator[Frame, None, None]:
 
     def close(self, code: Optional[int] = None, reason: Optional[str] = None) -> bytes:
         payload = bytearray()
-        if code is None and reason is not None:
+        if code is CloseReason.NO_STATUS_RCVD:
+            code = None
+        if code is None and reason:
```

Comment:
Could you change to `reason is not None`

Reply:
wsproto currently generates CloseConnection events with an empty string as the payload (https://github.com/python-hyper/wsproto/blob/38716a9e14dd5ff4512426567d40a09912e060fb/src/wsproto/frame_protocol.py#L529). Would you prefer if we change that to a `None` payload?